### PR TITLE
Small cleanup of diagnostics package

### PR DESCRIPTION
**Pull request by Donnerbart**
* fixed JavaDoc and comments format
* fixed some variable order (final vs. non-final, package-private etc.)
* fixed some warnings
* fixed `file.delete()` and its FindBugs suppression in `DiagnosticsLogFile`
  with `IOUtil.deleteQuietly()`
* used `addAll()` instead of foreach in `OverloadedConnectionsPlugin`
* made `DiagnosticsLogWriterImpl` package-private
* made `InvocationMonitor` a local variable in `OperationHeartbeatPlugin`
* removed unused method parameter in `StoreLatencyPlugin`
* removed deprecated `GroupProperty` from `SlowOperationPluginTest`

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/internal/diagnostics/DiagnosticsLogFile.java [hunk 1]
```diff
@@ -113,15 +113,13 @@ private PrintWriter newWriter() throws FileNotFoundException {
...
     private void rollover() {
         closeResource(printWriter);
         printWriter = null;
         file = null;
         index++;
 
         File file = new File(format(fileName, index - maxRollingFileCount));
-        // we don't care if the file was deleted or not
-        file.delete();
+        deleteQuietly(file);
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/internal/diagnostics/DiagnosticsLogFile.java]: My local FindBugs run is happy with this :)

File: hazelcast/src/main/java/com/hazelcast/internal/diagnostics/HealthMonitor.java [hunk 2]
```diff
@@ -180,9 +180,8 @@ private void logDiagnosticsHint() {
             // we only log the hint once
             performanceLogHint = false;
 
-            logger.info(String.format("The HealthMonitor has detected a high load on the system. For more detailed information,%s"
-                            + "enable the Diagnostics by adding the property -D%s=true",
-                    LINE_SEPARATOR, Diagnostics.ENABLED));
+            logger.info(format("The HealthMonitor has detected a high load on the system. For more detailed information,%n"
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/internal/diagnostics/HealthMonitor.java]: `%n` uses the system line separator

File: hazelcast/src/test/java/com/hazelcast/internal/diagnostics/SlowOperationPluginTest.java [hunk 3]
```diff
@@ -48,7 +47,6 @@ public void setup() {
         Config config = new Config()
                 .setProperty(SLOW_OPERATION_DETECTOR_ENABLED.getName(), "true")
                 .setProperty(SLOW_OPERATION_DETECTOR_THRESHOLD_MILLIS.getName(), "1000")
-                .setProperty(SLOW_INVOCATION_DETECTOR_THRESHOLD_MILLIS.getName(), "1000")
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/internal/diagnostics/SlowOperationPluginTest.java]: This was the last usage of this deprecated `GroupProperty`, so it really had no effect.

File: hazelcast/src/main/java/com/hazelcast/internal/diagnostics/OverloadedConnectionsPlugin.java [hunk 4]
```diff
@@ -198,13 +199,11 @@ private void clear() {
...
-     * @param q the queue to sample.
-     * @return the number of samples. If there were not sufficient samples, -1 is returned.
+     * @param q the queue to sample
+     * @return the number of samples (if there were not sufficient samples, -1 is returned)
      */
     private int sample(Queue<OutboundFrame> q) {
-        for (OutboundFrame frame : q) {
-            packets.add(frame);
-        }
+        packets.addAll(q);
```
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/internal/diagnostics/OverloadedConnectionsPlugin.java]: Was a suggestions from IDEA, which I already followed in the `EventQueuePlugin`.

File: hazelcast/src/main/java/com/hazelcast/internal/diagnostics/HealthMonitor.java (hunk 2 above)
Comment (pveentjer) [reply hazelcast/src/main/java/com/hazelcast/internal/diagnostics/HealthMonitor.java]: %n or \n ?

Comment (pveentjer) [reply hazelcast/src/main/java/com/hazelcast/internal/diagnostics/HealthMonitor.java]: I just did a small evaluation, and it seen %n workings like a line feed. I learned something new today.

Comment (Donnerbart) [reply hazelcast/src/main/java/com/hazelcast/internal/diagnostics/HealthMonitor.java]: Exactly. When using `String.format()` IDEA should suggest `%n` when `\n` is used. In our case we provided the correct `LINE_SEPARATOR`, but `%n` is just much shorter.

**Pull request merged**